### ISSUE TITLE
Gtk4 support

### ISSUE THREAD

**Issue by poldy** [giraffelibrary/giraffe#4]
First, thanks for the library, it's something I had been looking for.

I successfully got some simple Gtk3 apps to compile and run on an arm64 Mac (which required a bleeding-edge version of Poly/ML). However, I'd prefer to use version 4.x if possible.

This is just for personal interest, it's not a commercial application, so there's no hurry.

**Comment by pclayton** [comment]
Thanks for the feedback, in particular for confirming it works on an arm64 Mac, which I haven't been able to test yet.  I was concerned that recent verisons of Apple's GUI toolkit required the main thread to be used, causing an [issue with Poly/ML](https://github.com/polyml/polyml/issues/146), which runs application code in a worker thread.

Support for Gtk4 is certainly overdue.  I'll take a look at that once I have dealt the updates currently in progress.

**Comment by poldy** [comment]
No, your concerns were justified. I needed to use haoxuany's patch alright. I wrote up some details in a [blog post](https://portal.mozz.us/gemini/republic.circumlunar.space/users/dbane/gemlog/2022-11-27.gmi), but didn't think there was anything that justified a pull request to giraffe.
